consoles/VNC.pm

```diff
@@ -930,8 +935,8 @@ sub _receive_update {
 
     my $do_endian_conversion = $self->_do_endian_conversion;
 
-    foreach (1 .. $number_of_rectangles) {
-        $socket->read(my $data, 12) || die 'unexpected end of data';
+    foreach (my $i = 0; $i < $number_of_rectangles; ++$i) {
```

Comment:
note that this only works because number_of_rectangles is unsigned  :)

Reply:
I know, it is 0xFFFF.